crates/deltalake-core/src/data_catalog/mod.rs

```diff
@@ -25,11 +23,14 @@ pub enum DataCatalogError {
     Generic {
         /// Name of the catalog
         catalog: &'static str,
-
         /// Error message
         source: Box<dyn std::error::Error + Send + Sync + 'static>,
     },
 
+    /// Error from a specific catalog provider
+    #[error("Catalog implementation error: {0}")]
+    Error(Box<dyn std::error::Error + Send + Sync + 'static>),
+
```

Comment:
should integrations raise the above `Generic` error, which also contains an identifier from the specific catalogue? This pattern worked quite nicely in object store :).

Reply:
Can you link me to what you're referring to @roeap ? I can make the changes here once I understand what you're suggesting :smile:

Reply:
certainly :). https://github.com/delta-io/delta-rs/blob/2b913b37e71ed96212dcec8c3fc8e865754ced82/crates/deltalake-core/src/data_catalog/mod.rs#L25-L31